### ISSUE TITLE
Blur() not working properly.

### ISSUE THREAD

**Issue by peterkimga** [lovell/sharp#3425]
### Is this a possible bug in a feature of sharp, unrelated to installation?

- [x ] Running `npm install sharp` completes without error.
- [ x] Running `node -e "require('sharp')"` completes without error.

If you cannot confirm both of these, please open an [installation issue](https://github.com/lovell/sharp/issues/new?labels=installation&template=installation.md) instead.

### Are you using the latest version of sharp?

- [x ] I am using the latest version of `sharp` as reported by `npm view sharp dist-tags.latest`.

If you cannot confirm this, please upgrade to the latest version and try again before opening an issue.

If you are using another package which depends on a version of `sharp` that is not the latest, please open an issue against that package instead.

### What is the output of running `npx envinfo --binaries --system --npmPackages=sharp --npmGlobalPackages=sharp`?

  System:
    OS: macOS 12.6
    CPU: (8) arm64 Apple M1 Pro
    Memory: 255.53 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.15.1 - ~/.nvm/versions/node/v16.15.1/bin/node
    Yarn: 1.22.15 - ~/.nvm/versions/node/v16.15.1/bin/yarn
    npm: 8.11.0 - ~/.nvm/versions/node/v16.15.1/bin/npm
  npmPackages:
    sharp: ^0.31.1 => 0.31.1 

### What are the steps to reproduce?

Having 2048 on 2048 px PNG of a gradient and using it with the below code I expect the outcome to be a smooth gradient but it's not.

### What is the expected behaviour?

![image](https://user-images.githubusercontent.com/57945998/199242439-c5b12a3e-fe58-45e5-ab1c-eaabcc56fba5.png)

but instead, my output looks like this:

![image](https://user-images.githubusercontent.com/57945998/199243011-2c9ba45d-d569-4e77-80e8-922c0fe939fe.png)

### Please provide a minimal, standalone code sample, without other dependencies, that demonstrates this problem

Input PNG file "gradient.png":

![image](https://user-images.githubusercontent.com/57945998/199243243-629dfb81-71b6-4dc4-850d-25b191e86601.png)

and the code:

```
	const extract = {
		left: 1024 - 64,
		top: 1024 - 64,
		width: 128,
		height: 128,
	};

	sharp("gradient.png")
		.extract(extract)
		.normalise()
		.blur(15)
		.grayscale(true)
		.toColourspace('b-w')
		.png({
			compressionLevel: 9,
			adaptiveFiltering: true,
			force: true,
		})
		.toFile(`gradient_new.png`, function(err) {
			if(err) {
				LogError.show(`Error`, err);
			}
		});
```

### Please provide sample image(s) that help explain this problem

When the code reaches normalize step all looks as expected:

```
.normalise()
```
and after that the image levels histogram looks like this:

![image](https://user-images.githubusercontent.com/57945998/199244745-07f8ee52-18a7-45b8-9b26-d8796d9eb118.png)

But when blurring it in gimp:

![image](https://user-images.githubusercontent.com/57945998/199243931-bc39c602-49bd-4429-8db9-7b356363a2d7.png)

the histogram is filled with colors to fill the gaps:

![image](https://user-images.githubusercontent.com/57945998/199244038-8ba282c9-84ef-48d7-acf7-75b73c5dfdbe.png)

But the code does not do it, it just smooths the edges and creates ripples of color instead of adding new ones:

![image](https://user-images.githubusercontent.com/57945998/199243802-a1792557-af31-49ac-b114-75328c6a2d2b.png)

I think the Blur is not doing what it should do. It would be nice to have smooth color after that.

Regards
Peter

**Comment by peterkimga** [comment]
@lovell Hi

Thanks for the quick replay. I will tell you that working with your class is one of the most intuitive I ever used so far and this issue was something that I did not expect. I used another package just to blur images again to keep me going. I do not like to mix packages just for 1 thing to overcome the problem so I'm glad that you responded.

Your solution did work, but I do think it breaks the experience of piping the operations that is super intuitive in your package.

Will you try to make it run as expected? Cause this is clearly a temporary solution?

Regards
Peter
